### ISSUE TITLE
topsites entry locked on waterfox alpha

### ISSUE THREAD

**Issue by smahat96** [BrowserWorks/waterfox#979]
the top sites in the newtab page were missing and also missing from the settings and when I tried to edit "browser.newtabpage.activity-stream.feeds.topsites" in about:config .it was locked !
 and even when editing the pref.js from the profile directory it didn't work .was this on purpose ??

**Comment by grahamperrin** [comment]
Yeah, I don't like to criticise but this type of thing is **not** focused on "[giving users choice](https://www.waterfox.net/about/)": 

![image](https://user-images.githubusercontent.com/192271/57990251-b617a300-7a9c-11e9-972c-c7623da2e2c5.png)

An oversight, maybe? (I can understand; it's an early alpha.)

Or is this a consequence of denying users of Pocket the choice to have saved pages shown in _Highlights_?

**Comment by smahat96** [comment]
could be an oversight and if there is no giude or a way of fixing it myself or I hope it will be fixed in next release.

**Comment by grahamperrin** [comment]
/me looks at the alpha code for the first time 

https://github.com/MrAlex94/Waterfox/blob/f0b4a500c4d0e307866df3172015cc3257263d7e/browser/app/profile/firefox.js#L1200

Speeding lazily through the file history I don't see a point at which it was locked. (Not a problem. Just curious.)

**Comment by smahat96** [comment]
well  since the developer hasn't responded to the issue I have no idea if he saw and if he is going to change it in next release ..i guess we have to wait and see

**Comment by laniakea64** [comment]
I briefly looked into some aspects of the new tab page in v68, including this, and I don't see what's so anti-user about "top sites" feature that it deserves to be hard-disabled.  All I noticed when unlocking the pref and enabling it was the addition of some links to sites + image tiles for said links, as well as one-click access to some installed search engines.  And the sites/search engines that are displayed can be easily customised, AFAICT nothing is being forced or pushed.

This looks useful and again I don't see how this feature is anti-user.  What am I missing?

Hi @MrAlex94 Did you intend to lock `browser.newtabpage.activity-stream.feeds.section.topstories` ("Recommended by Pocket") instead?

**Comment by laniakea64** [comment]
> Speeding lazily through the file history I don't see a point at which it was locked.

Apparently top sites is "privacy-unfriendly" - https://github.com/MrAlex94/Waterfox/commit/1983d09d4f35d6d9e58f8088b55018e40525feb4#diff-d7a0401e94f68887594ad1d1d9f9c28dR1182

What specifically is privacy-unfriendly about it?

**Comment by MrAlex94** [comment]
Yes was definitely an oversight. Will be fixed.

**Comment by laniakea64** [comment]
Thanks @MrAlex94 for fixing this in https://github.com/MrAlex94/Waterfox/commit/091295d9dc29b66988f5ace8d656b3a4b1685948 ! :smiley: 

In addition to the pref having been locked, the options UI for top sites was removed too, I believe that was done in https://github.com/MrAlex94/Waterfox/commit/28ecee473cb2e855ea116fe8f36342ca51ca486c#diff-ca12ba81c967dce1ca60c6df0494ae4f but not sure if that's the only related commit.
@MrAlex94 Could you please also restore the options UI for top sites?

**Comment by grahamperrin** [comment]
**about:config?filter=browser.newtabpage.activity-stream.feeds.topsites**

Tested with 68.0a2 (20190801144413) on Windows. Both preferences default to true, and can be toggled: 

![image](https://user-images.githubusercontent.com/192271/62427391-608d8280-b6ea-11e9-800d-3717176e17c0.png)

Preferences GUI (after reverting to defaults at `about:config`): 

![image](https://user-images.githubusercontent.com/192271/62427447-e01b5180-b6ea-11e9-8462-89dd3e229551.png)

![image](https://user-images.githubusercontent.com/192271/62427417-97fc2f00-b6ea-11e9-95b7-e6b4e9e0d8fa.png)

**Comment by smahat96** [comment]
Thanks @MrAlex94 for fixing it

**Comment by laniakea64** [comment]
For reference, could someone please point to where https://github.com/MrAlex94/Waterfox/issues/979#issuecomment-505255601 was addressed?  Thanks.

(I haven't noticed anything related to that comment in the [commit log](https://github.com/MrAlex94/Waterfox/commits/gecko68); and my current 68.0a2 build - which doesn't have the options UI for top sites - is now somewhat outdated and I haven't been able to find the time to do a new build.)

**Comment by grahamperrin** [comment]
@laniakea64 (sorry) I don't understand the question …

**Comment by laniakea64** [comment]
Maybe a screenshot of **Firefox** 68.0.2 `about:preferences` will help clarify? -

![Screen shot 2019-08-23 1](https://user-images.githubusercontent.com/16628411/63562587-2ca8be80-c52c-11e9-9388-0c24f5e7f88c.png)

I outlined in red what I mean by "the options UI for top sites".  This doesn't exist in my (older) Waterfox 68.0a2.

As this issue is now closed, I'm wondering:
 - What commit restored this UI to Waterfox 68?
 - Or if it hasn't been restored, did @MrAlex94 comment somewhere saying that he decided not to restore it and why not?
